replica/table.cc

```diff
@@ -512,6 +512,13 @@ std::vector<lw_shared_ptr<compaction_group>> table::compaction_groups() const {
     return ret;
 }
 
+future<> table::for_each_compaction_group(std::function<future<>(compaction_group&)> action) {
+    auto cgs = compaction_groups();
+    for (compaction_group_ptr& cg : cgs) {
+        co_await action(*cg);
+    }
+}
```

Comment:
(here we can insert the barriers later once we have dynamic compaction groups)

Reply:
I think I'll rename it to parallel_foreach_compaction_group(). Parallelism is really important as you want all groups to make progress on some requested table-wide operation, even if they're serialized down the road, like in compaction manager. For operations that don't need parallelism, we can go with a simple loop over replica::table::compaction_groups().